src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,38 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            return new BulldozerProducer().get();
+        } else if (type == Excavator.class) {
+            return new ExcavatorProducer().get();
+        } else if (type == Truck.class) {
+            return new TruckProducer().get();
+        }
+        return new ArrayList<>();
```

Comment:
Instead of returning null or a specific type list, it's better to return an empty list when no producer has been found. This is already correctly implemented with `return new ArrayList<>();`.